open-apparel-registry/open-apparel-registry
# Enable submitting claim a facility form

## Overview

This PR enables users to complete and submit a facility claim form via a new route in the UI. That route currently lives behind a feature flag -- and we render the map as a fallback if the correct flag is not active.

The workflow is:

- user clicks on a link or a badge in the facility details page
- user goes to a `facilities/id/claim` route
- if the user is not signed in, we show a link to the login page
- if the user is signed in, we render the wizard from from the wireframes
- the form validates the email address and company name as required fields. Everything else is optional
- submitting the form creates a new `FacilityClaim` object in the API and sends an email to the user to confirm that the claim has been submitted.

There is some remaining work here but I wanted to get some feedback on it now. The remaining work I see here is:

- prevent users who have already submitted a claim on a facility from re-submitting the form for that facility (this is now #555)
- ~render error messages from the API in the form wizard~
- ~add some kind of "end state" to the form wizard to indicate that the user has successfully submitted the form~
- ~clean up the form style~

Connects #515
Connects #544 

## Demo

![Screen Shot 2019-06-03 at 3 53 13 PM](https://user-images.githubusercontent.com/4165523/58830345-1e779000-8618-11e9-8e7a-b7094e14beb3.png)

![Screen Shot 2019-06-03 at 3 53 30 PM](https://user-images.githubusercontent.com/4165523/58830356-26373480-8618-11e9-8940-1b8809c3c755.png)

![Screen Shot 2019-06-03 at 3 53 36 PM](https://user-images.githubusercontent.com/4165523/58830362-29cabb80-8618-11e9-841f-7a971bd697d0.png)

![Screen Shot 2019-06-03 at 3 53 43 PM](https://user-images.githubusercontent.com/4165523/58830375-2d5e4280-8618-11e9-9d53-e778ce6e9c36.png)

![Screen Shot 2019-06-03 at 3 53 58 PM](https://user-images.githubusercontent.com/4165523/58830385-35b67d80-8618-11e9-8ca5-b6368c81fb97.png)

![Screen Shot 2019-06-03 at 3 54 03 PM](https://user-images.githubusercontent.com/4165523/58830391-38b16e00-8618-11e9-98a8-e304ea452733.png)

## Testing Instructions

- get this branch then `./scripts/update`, `./scripts/manage resetdb`, `./scripts/manage createsuperuser` and `./scripts/server`
- visit localhost:6543 and search for a facility. from the details page, add the `/claim` path after the OAR ID and verify that it does not render the claim form (since the feature flag is still off)
- sign in to the django-admin and turn the feature flag on
- refresh the claim path on localhost:6543 and verify that you now see the claim form, since you are signed in
- sign out, then sign in as an account with a contributor attached (c8@example) then return to the claim form
- complete the form and submit it, verifying that the submission completes, that you reach the confirmation step of the wizard, and that you see the confirmation email sent in the terminal

## Checklist
